src/cephadm/cephadm.py

```diff
@@ -4751,6 +4753,32 @@ def command_gather_facts(ctx: CephadmContext) -> None:
 ##################################
 
 
+def command_custom_logrotate(ctx: CephadmContext) -> str:
+    """Writes out custom logrotate config for either cephadm.log or a cluster's cluster logs"""
+    custom_template: str = ''
+    if 'logrotate_config_file' in ctx and ctx.logrotate_config_file:
+        logger.info('Grabbing logrotate config from filepath...')
+        try:
+            with open(ctx.logrotate_config_file) as f:
+                custom_template = f.read()
+        except FileNotFoundError:
+            raise Error(f'Failed to find logrotate config at {os.path.abspath(ctx.logrotate_config_file)}')
+    else:
+        logger.info('Grabbing logrotate config from stdin...')
+        custom_template = sys.stdin.read()
+
+    if 'cluster' in ctx and ctx.cluster:
+        if not ctx.fsid:
+            raise Error('--cluster requires also passing the FSID with --fsid <FSID>')
+        write_cluster_logrotate_config(ctx, ctx.fsid, custom_template, overwrite=True)
+    else:
+        write_cephadm_logrotate_config(ctx, custom_template, overwrite=True)
+
+    return 'Successfully wrote out logrotate config'
```

Comment:
```
2025-04-01T22:40:46.828004+0000 mgr.smithi203.uuthde (mgr.14231) 8 : cephadm [ERR] Failed to write cluster custom logrotate config to smithi007 | 1 | [''] | ['Grabbing logrotate config from stdin...\nSuccessfully wrote out logrotate config'] 
```
this is causing this to be treated as a non-zero rc. We can't return a string from here